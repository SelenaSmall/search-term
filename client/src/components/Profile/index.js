import React from 'react';
import {
 Badge, Col, Row, Progress 
} from 'reactstrap';
import styled from 'styled-components';

const StepContainer = styled.div`
  width: 100%;
  display: flex;
  &:after {
    content: '';
    position: relative;
    flex: 1;
    margin: 0.8rem 0 0 2rem;
    height: 1px;
    background-color: rgba(0, 0, 0, 0.1);
  }
`;
const BadgeContainer = styled.span`
  margin-right: 0.5em;
`;
const StepSpan = styled.span`
  margin-top: 0.1em;
`;

const Step = ({ number, done, name }) => (
  <Col>
    <StepContainer data-test-id={`step-${number}`}>
      <BadgeContainer>
        <Badge color={done ? 'primary' : 'secondary'} pill>
          {number}
        </Badge>
      </BadgeContainer>
      <StepSpan>{name}</StepSpan>
    </StepContainer>
  </Col>
);

export default () => (
  <>
    <Row data-test-id="profile-steps">
      <Step number="1" done name="Id" />
      <Step number="2" name="Alias" />
      <Step number="3" name="Email" />
      <Step number="4" name="Avatar" />
      <Step number="5" name="Confirmation" />
    </Row>
    <Row>
      <Col>
        <Progress value="20" />
      </Col>
    </Row>
  </>
);
