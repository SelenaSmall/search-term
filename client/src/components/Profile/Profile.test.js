import React from 'react';
import { shallow } from 'enzyme';
import Profile from ".";

it('renders without crashing', () => {
  const wrapper = shallow(<Profile />);
  expect(wrapper).toMatchSnapshot();
});
