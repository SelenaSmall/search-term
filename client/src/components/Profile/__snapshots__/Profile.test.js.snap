// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <Row
    data-test-id="profile-steps"
    tag="div"
  >
    <Step
      done={true}
      name="Id"
      number="1"
    />
    <Step
      name="Alias"
      number="2"
    />
    <Step
      name="Email"
      number="3"
    />
    <Step
      name="Avatar"
      number="4"
    />
    <Step
      name="Confirmation"
      number="5"
    />
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Progress
        max={100}
        tag="div"
        value="20"
      />
    </Col>
  </Row>
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Form
        tag="form"
      >
        <FormGroup
          tag="div"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Alias
          </Label>
          <Input
            name="alias"
            placeholder="alias to be seen by everyone"
            type="text"
          />
        </FormGroup>
        <Button
          color="primary"
          tag="button"
          type="submit"
        >
          Next
        </Button>
      </Form>
    </Col>
  </Row>
</Fragment>
`;
